src/pinns_pde_solve.jl

```diff
@@ -1,5 +1,5 @@
 import Base.Broadcast
-# Base.Broadcast.dottable(x::Function) = true
+Base.Broadcast.dottable(x::Function) = true
```

Comment:
to use @register, I needed this again

Reply:
https://github.com/SciML/NeuralPDE.jl/issues/353

Reply:
`ModelingToolkit.@register`?

Reply:
yes

Reply:
Can you show a minimal example of what it's needed for now?

Reply:
```julia
@parameters θ
@variables u(..)
Dθ = Differential(θ)
fx(θ) = θ
@register fx(θ)

eq = Dθ(u(θ)) ~ 2*fx(θ)

expr =ModelingToolkit.toexpr(eq.rhs)
```


```julia
julia> dump(expr)
Expr
  head: Symbol call
  args: Array{Any}((3,))
    1: * (function of type typeof(*))
    2: Int64 2
    3: Expr
      head: Symbol call
      args: Array{Any}((2,))
        1: fx (function of type typeof(fx))
        2: Symbol θ
```
`fx` is a function inside `expr` if we convert it in the symbol(in the main branch), 
@register function is not working and can't be called in an inner representation 
as a function because it converts to a symbol.
Convert in symbol was done for avoid broadcast piracy `Base.Broadcast.dottable(x::Function) = true`

https://github.com/SciML/NeuralPDE.jl/blob/d42dbd54072567096407e990c8685a63ea7d1b13/src/pinns_pde_solve.jl#L188-L190

here broadcast avoid function
```julia
Base.Broadcast.dottable(x::Function) = false

julia> Broadcast.__dot__(expr)
:((*)(2, (fx)(θ)))
```
here functions aslo broadcasted
```julia
Base.Broadcast.dottable(x::Function) = true

Broadcast.__dot__(expr)
:((*).(2, (fx).(θ)))
```
so if we want to use ModelingToolKit@register and broadcast work correct is need `Base.Broadcast.dottable(x::Function) = true`

and macro @. only works with direct insert.
Maybe some trick is needed here which I don't know

```julia
julia>  expr
:((*)(2, (fx)(θ)))

julia> @. expr
:((*)(2, (fx)(θ)))

julia> @. :((*)(2, (fx)(θ)))
:((*).(2, fx.(θ)))
```